webapp/src/pages/PlayPage/PlayPage.tsx

```diff
@@ -43,8 +47,17 @@ export default function PlayPage() {
     {
       variant: ButtonVariant.SUBMIT,
       iconVariant: IconVariant.ADD,
-      onClick: () => navigate(PLAY_GAME_URL),
-      children: 'New game',
+      onClick: () => {
+        socket.emit(gameQueueClientToServerWsEvents.gameQueueJoin);
+        setGameCtx &&
+          setGameCtx({
+            isPlaying: false,
+            isWaitingToPlay: true,
+            gameRoomId: null,
+          }); // TODO: these setGameCtx should be broadcasted to all tabs
+        navigate(PLAY_GAME_QUEUE);
```

Comment:
Since this button does more things than just navigating, I think it's better to keep it here in its kind of final form, and put some hardcoded link somewhere else to navigate to the provisional PLAY_GAME_URL.